dominos/src/main/java/me/fontys/semester4/dominos/configuration/data/store/StoreImporter.java

```diff
@@ -0,0 +1,108 @@
+package me.fontys.semester4.dominos.configuration.data.store;
+
+import me.fontys.semester4.data.entity.Store;
+import me.fontys.semester4.data.repository.StoreRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.io.Resource;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.*;
+
+@Configuration
+public class StoreImporter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StoreImporter.class);
+
+
+    @Qualifier("stores")
+    private final Resource[] stores;
+    private final StoreRepository storeRepository;
+    private final Map<String, Integer> warnings = new HashMap<>();
+
+    @Autowired
+    public StoreImporter(
+            Resource[] stores,
+            StoreRepository storeRepository
+    ) {
+        this.stores = stores;
+        this.storeRepository = storeRepository;
+    }
+
+    public void doImport() throws IOException {
+        LOGGER.info("Starting import of stores...");
+
+        this.warnings.clear();
+        List<Store> stores = new ArrayList<>();
+
+        for (Resource resource : this.stores) {
+            LOGGER.info("Reading resource " + resource.getFilename());
+            stores.addAll(this.processStoreResource(resource));
+        }
+
+        LOGGER.info(String.format("Inserting %s stores...", stores.size()));
+        this.storeRepository.saveAll(stores);
+    }
+
+    public void report() {
+        int totalWarnings = this.warnings.values().stream()
+                .reduce(0, Integer::sum);
+
+        if (totalWarnings > 0) {
+            LOGGER.warn(String.format("Store import result : %s warnings", totalWarnings));
+
+            for (Map.Entry<String, Integer> warning : this.warnings.entrySet()) {
+                LOGGER.warn(String.format("  -> %s : %s", warning.getKey(), warning.getValue()));
+            }
+        }
+    }
+
+    private List<Store> processStoreResource(Resource resource) throws IOException {
+        BufferedReader in = new BufferedReader(new InputStreamReader(resource.getInputStream()));
+
+        List<String> allStrings = new ArrayList<String>();
```

Comment:
Not really a needed fix, but just FYI;
```List<String> allStrings = new ArrayList<String>();```
Is unnecessary and can be replaced with;
```
List<String> allStrings = new ArrayList<>();
```
You don't have to define the String generic twice here. IntelliJ should also mark the 2nd String keyword in gray here. Just a tip